File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> check = new HashSet<>();
        int left = 0;
        int maxLength = 0;
        for(int right=0; right<s.length(); right++){
            char current = s.charAt(right);
            while(check.contains(current)){
                check.remove(s.charAt(left));
                left++;
            }
            check.add(current);
            maxLength = Math.max(maxLength, right-left+1);
        }

        return maxLength;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0 or len(s) == 1:
            return len(s)

        left = 0
        right = 0
        longest = 0
        check = set()
        while right<len(s):
            while s[right] in check:
                check.remove(s[left])
                left += 1

            check.add(s[right])
            longest = max(longest,right-left+1)
            right += 1
        return longest
            